### ISSUE TITLE
How to add a strongly typed SignalR Hub when using Mangament

### ISSUE THREAD

**Issue by ZisisTsatsas** [Azure/azure-signalr#1775]
### Can I add a strongly typed SignalR Hub when using [Management](https://github.com/Azure/azure-signalr/blob/dev/docs/management-sdk-guide.md) on a serverless configuration?

I want to add a strongly typed hub in my signalr service to be used by the client, is this possible when using Management and ServiceHubContext?

If not how can invoke methods from clients? Send messages between clients or have clients add themselves to groups for example.

**Comment by Y-Sindo** [comment]
Although strongly typed hub is supported, client invocation is not supported yet.  Duplicated: #1756

**Comment by ZisisTsatsas** [comment]
is there plans for it to be supported?

**Comment by Y-Sindo** [comment]
> is there plans for it to be supported?

Yes

**Comment by Y-Sindo** [comment]
@ZisisTsatsas  Sorry that I have found that I may misunderstand your problem. What you want to do in [Call hub methods from clients](https://learn.microsoft.com/aspnet/core/signalr/dotnet-client?view=aspnetcore-7.0&tabs=visual-studio#call-hub-methods-from-client), right? It's not convenient to achieve it only with Management SDK. We recommend you use Azure Functions with [SignalR trigger](https://learn.microsoft.com/azure/azure-functions/functions-bindings-signalr-service-trigger?tabs=in-process&pivots=programming-language-csharp#with-class-based-model).

* Here is an example: 
https://github.com/aspnet/AzureSignalR-samples/tree/main/samples/BidirectionChat

*To use Azure Functions with SignalR trigger, you use Azure SignalR Service in serverless mode. To understand how serverless mode work, see https://learn.microsoft.com/en-us/azure/azure-signalr/concept-service-mode#serverless-mode

* The following docs should give you an overview about how to use SignalR service in Azure functions https://learn.microsoft.com/en-us/azure/azure-signalr/signalr-concept-azure-functions
https://learn.microsoft.com/en-us/azure/azure-signalr/signalr-concept-serverless-development-config

Please note that [Strongly typed hubs in SignalR](https://learn.microsoft.com/aspnet/core/signalr/hubs?view=aspnetcore-7.0#strongly-typed-hubs) is more of the data flow from server to client.

**Comment by ZisisTsatsas** [comment]
@Y-Sindo Is there a way to do this without Azure Functions? Currently, I am hosting SignalR alongside my API in an Azure WebApp and I am using Redis Stack to have consistency over different regions, but it has some issues so I want to switch to the Azure SignalR service

**Comment by Y-Sindo** [comment]
Yes, you can do this without Azure Functions. SignalR trigger is nothing more than a REST API server that handles client events. Therefore, you can also implement a REST API server to do this:
* See the REST API spec here: https://learn.microsoft.com/en-us/azure/azure-signalr/concept-upstream#serverless-protocols
* Additionally, the response body of invocation message can be a [CompletionMessage](https://github.com/dotnet/aspnetcore/blob/82ff9246592477abf4ff57ac905a368cbd99ff20/src/SignalR/common/SignalR.Common/src/Protocol/CompletionMessage.cs) in the SignalR protocol, which is serialized with a hub protocol used by your clients. Only JSON hub protocol and message pack hub protocol are supported. You can see the [SignalR trigger code](https://github.com/Azure/azure-sdk-for-net/blob/63479f872ba91ab5afa4a9fea426dbb45ad67054/sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/src/TriggerBindings/Executor/SignalRInvocationMethodExecutor.cs#L42-L67) for an example to construct the HTTP response message for invocation message.
